backend/dataline/services/conversation.py

```diff
@@ -166,3 +171,21 @@ async def query(
             message=MessageOut.model_validate(stored_ai_message),
             results=serialized_results,
         )
+
+    async def get_conversation_history(self, session: AsyncSession, conversation_id: UUID) -> list[BaseMessage]:
+        """
+        Get the last 10 messages of a conversation (AI, Human, and System)
+        """
+        messages = await self.message_repo.get_by_conversation(session, conversation_id)
+        base_messages = []
+        for message in messages:
+            if message.role == BaseMessageType.HUMAN.value:
+                base_messages.append(HumanMessage(content=message.content))
+            elif message.role == BaseMessageType.AI.value:
+                base_messages.append(AIMessage(content=message.content))
+            elif message.role == BaseMessageType.SYSTEM.value:
+                base_messages.append(SystemMessage(content=message.content))
+            else:
+                logger.error(Exception(f"Unknown message role: {message.role}"))
+
+        return base_messages[-10:]
```

Comment:
Very nitpicky, but I think it's better to get the last 10 messages first and loop over them, rather than loop over all the messages and then return the last 10 (especially since we're not discarding any of the messages)

Reply:
Yeahh I thought about it but just kept coding at the time cause no one's gonna have big messages and it's all local so idc much. But yeah would be better to have a limit on this. Maybe we can add a TODO for the future